### DISCUSSION TITLE
Lanbon L8 - GPIO "off" doesnt work

### DISCUSSION THREAD

**Discussion by freshnas** [HASwitchPlate/openHASP#289] [General]
Hello,

I am using firmware 6.2 and my lanbon l8 is connected to iobroker via mqtt.
i can turn on the GPIOs using output12,output14,output27...33 ={"state": "on"} but i can't turn it off with {"state": "off"}  .
i tried the GPIO config [197658,263456,329249,655628,655886,656155,0,0] and [721164,721422,197658,721691,263456,329249,0,0] but i can't get the GPIOs off anymore.

![image](https://user-images.githubusercontent.com/83093888/150750250-6d0ae0a1-bc23-4b2a-8088-00fa9066114d.png)

**Comment by fvanroie** [comment]
Can you check the telnet logs when the `off` message is sent?
Also, try to send the `off` message from another mqtt client like `mosquitto_pub` or `MQTT Explorer`?

Without the logs, it is hard to know what is happing or the issue might be..

**Reply by freshnas** [reply]
hello, thanks for the answer.

For the test with telnet log I used the MQTT explorer.
This is the result:

```
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state":"0"}
MQTT PUB: output12 => {"state":"on"}
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state":"0"}
MQTT PUB: output12 => {"state":"on"}
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state":"off"}
MQTT PUB: output12 => {"state":"on"}
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state":"off"}
MQTT PUB: output12 => {"state":"on"}
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state":"false"}
MQTT PUB: output12 => {"state":"on"}
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state":"false"}
MQTT PUB: output12 => {"state":"on"}
```

**Comment by freshnas** [comment]
addendum:

with [false] it works

```
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state": "on"}
MSGR: GROUP 1 value 1 (0-1)
MQTT PUB: output12 => {"state":"on"}
MQTT RCV: hasp/lanbon_l8/command/output12 = {"state": false}
MSGR: GROUP 1 value 0 (0-1)
MQTT PUB: output12 => {"state":"off"}
```

**Comment by fvanroie** [comment]
@fvanroie said in [GPIO ports for the grove connector?](https://community.m5stack.com/topic/3918/gpio-ports-for-the-grove-connector/16):
> So I'm left thinking there is a bug in the command parser or the documentation isn't clear...

I can confirm that this problem is a **parser bug.** It currently only accepts *unquoted* boolean values of `true/false` or `1/0` instead of the usual range of: `on/off`, `true/false`, `0/1`, `yes/no` which might be quoted or not.
 
This is particularly confusing since the state is always reported as `on/off`.
 
Thank you very much for bringing this issue to my attention. I have pushed a fix to GitHub and it is available for testing from [the Actions tab](https://github.com/HASwitchPlate/openHASP/actions/runs/1770884100) and updated [the documentation](https://haswitchplate.github.io/openHASP-docs/0.6.2/commands/#outputx).
